onnxruntime/contrib_ops/cpu/bert/qembed_layer_norm.cc

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "qembed_layer_norm.h"
+
+#include <cmath>
+
+#include "embed_layer_norm_helper.h"
+#include "core/framework/op_kernel.h"
+#include "core/providers/common.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+namespace {
+
+// TODO(kreeger): Find a global home for this helper method.
+template <typename T>
+inline float Dequantize(T value, float scale, T zero_point) {
+  return static_cast<float>(static_cast<int32_t>(value) - zero_point) * scale;
+}
+
+}  // namespace
+
+// This op is internal-only, so register outside of onnx:
+#define REGISTER_KERNEL_TYPED(T)                                  \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                  \
+      QEmbedLayerNormalization,                                   \
+      kMSDomain,                                                  \
+      1,                                                          \
+      T,                                                          \
+      kCpuExecutionProvider,                                      \
+      KernelDefBuilder()                                          \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
```

Comment:
You need to differentiate int8 and uint8 if you want to support both

Reply:
Done - first pass at supporting both w/o a lookup table. As mentioned I'll tackle these optimizations in follow up PRs that I can provide documentation to future contributors.